pkg/provider/azure_controller_common.go

```diff
@@ -663,3 +711,25 @@ func isInstanceNotFoundError(err error) bool {
 	}
 	return strings.Contains(errMsg, errStatusCode400) && strings.Contains(errMsg, errInvalidParameter) && strings.Contains(errMsg, errTargetInstanceIds)
 }
+
+func getErrorCode(err error) *string {
```

Comment:
I think it's better returning `string` instead of `*string`, returning pointer has risk of panic.
And write comments and ut for this func.

Reply:
`getAzureErrorCode` is better since it's azure specifc